src/components/HistoryTable/HistoryTableRow/FoldedCommit.tsx

```diff
@@ -16,56 +24,68 @@ interface FoldedCommitProps {
   toggleRowSize: (idx: number, numCommits: number) => void;
   numVisibleCols: number;
   style?: CSSProperties;
+  selected: boolean;
 }
-export const FoldedCommit: React.FC<FoldedCommitProps> = ({
-  index,
-  rolledUpCommits,
-  toggleRowSize,
-  numVisibleCols,
-  style,
-}) => {
-  // The commits are expanded if the height of the element is not equal to FOLDED_COMMITS_HEIGHT.
-  const { height } = style;
-  const defaultOpen = height !== FOLDED_COMMITS_HEIGHT;
-  const numCommits = rolledUpCommits.length;
+export const FoldedCommit = memo(
+  ({
+    index,
+    rolledUpCommits,
+    toggleRowSize,
+    numVisibleCols,
+    style,
+    selected,
+  }: FoldedCommitProps) => {
+    // The commits are expanded if the height of the element is not equal to FOLDED_COMMITS_HEIGHT.
+    const { height } = style;
 
-  const columns = Array.from(Array(numVisibleCols)).map((_, idx) => (
-    <EmptyCell key={`loading_row_${idx}`} /> // eslint-disable-line react/no-array-index-key
-  ));
+    // The virtualized table will unmount the row when it is scrolled out of view but it will cache its height in memory.
+    // This means we can't rely on the state of the row to determine if it is expanded or not.
+    // So we instead look at its height which is cached by the table and determine if it is expanded or not.
```

Comment:
afaict resetAfterIndex tells react-window to recalculate all of the rows including the heights. Since we rely on the row height to determine the state of this component we store its value in the context so we have a guarantee that its state will be preserved despite how react-window decides to process it.